### PR TITLE
Fix settings screen layout on iOS and add soft back button

### PR DESCRIPTION
**Pull request by zbettenbuk**

### CODE REVIEW

File: react/features/app-settings/components/AppSettings.native.js [hunk 1]
```diff
@@ -39,7 +41,11 @@ class AppSettings extends AbstractAppSettings {
                 presentationStyle = 'fullScreen'
                 style = { styles.modal }
                 visible = { this.props._visible }>
+                <StatusBar hidden = { true } />
```
Comment (saghul) [inline react/features/app-settings/components/AppSettings.native.js]: Why do we want this though? I think we shouldn't hide it.

**Comment by saghul** [comment]
There needs to be a top margin equal to the one on the welcome page, otherwise it's broken for the iPhone X:

screen shot 2018-01-08 at 13 27 14

**Comment by saghul** [comment]
Also, the settings page doesn't seem to handle rotation, whereas the welcome page does.

**Comment by zbettenbuk** [comment]
@saghul The settings screen now uses a custom implementation of paddings around header and content minimising the blank area on every phone, also focusing on iPhone X. It doesn't support the Essential Phone as I didn't have a method to detect it and nor device to test. But I think we can live with that for now, especially as the top notch of that will likely not interfere with the "Settings" label, being quite narrow.

In long term we need a more generic solution as we discussed, but I'm not a big fan of anything that renders too much static-empty areas where not really needed.

**Comment by zbettenbuk** [comment]
@saghul Also: I fixed the crash around the URL validation, and removed the overuse of component state in the abstract logic (hence the removed type and some functions).

**Comment by zbettenbuk** [comment]
image
image

File: react/features/app-settings/components/AbstractAppSettings.js [hunk 2]
```diff
@@ -304,8 +173,19 @@ export class AbstractAppSettings extends Component<Props, State> {
  * @returns {Object}
  */
 export function _mapStateToProps(state: Object) {
+    const _configDefaultURL = state['features/app'].app._getDefaultURL();
+    const _profile = getProfile(state);
+
+    console.log('ZB ::: New profile', JSON.stringify(_profile));
+
+    if (typeof _profile.defaultURL !== 'string') {
```
Comment (saghul) [inline react/features/app-settings/components/AbstractAppSettings.js]: why do we need this? why not leave it as undefined? right now it shows in "black" as if the user typed it, IMHO it should remain undefined.

File: react/features/app-settings/components/AppSettings.native.js [hunk 3]
```diff
@@ -23,6 +28,17 @@ import { translate } from '../../base/i18n';
  */
 class AppSettings extends AbstractAppSettings {
 
+    /**
```
Comment (saghul) [inline react/features/app-settings/components/AppSettings.native.js]: style: unneeded extra newline

File: react/features/app-settings/components/AppSettings.native.js [hunk 4]
```diff
@@ -36,42 +52,49 @@ class AppSettings extends AbstractAppSettings {
             <Modal
                 animationType = 'slide'
                 onRequestClose = { this._onRequestClose }
-                presentationStyle = 'fullScreen'
-                style = { styles.modal }
+                presentationStyle = 'overFullScreen'
```
Comment (saghul) [inline react/features/app-settings/components/AppSettings.native.js]: Why is this needed? According to the docs it's the same as fullScreen, plus transparency, which we are not using: https://facebook.github.io/react-native/docs/modal.html#presentationstyle It also seems to be the default, so can we just remove the prop altogether?

File: react/features/app-settings/components/AppSettings.native.js [hunk 5]
```diff
@@ -36,42 +52,49 @@ class AppSettings extends AbstractAppSettings {
...
+                supportedOrientations = { [
+                    'portrait',
+                    'landscape'
```
Comment (saghul) [inline react/features/app-settings/components/AppSettings.native.js]: style: alpha-sort

File: react/features/app-settings/components/AppSettings.native.js [hunk 6]
```diff
@@ -36,42 +52,49 @@ class AppSettings extends AbstractAppSettings {
...
                         fieldSeparator = { true }
                         i18nLabel = 'profileModal.serverURL' >
                         <TextInput
                             autoCapitalize = 'none'
                             onChangeText = { this._onChangeServerName }
-                            onEndEditing = { this._onSaveServerName }
-                            placeholder = 'https://jitsi.example.com'
-                            value = { this.state.serverURL } />
+                            placeholder = { this.props._configDefaultURL }
+                            value = { this.props._profile.defaultURL } />
```
Comment (saghul) [inline react/features/app-settings/components/AppSettings.native.js]: we seem to have a mix of serverURL and defaultURL. Can we make that consistent?

File: react/features/app-settings/components/AppSettings.native.js [hunk 7]
```diff
@@ -80,20 +103,43 @@ class AppSettings extends AbstractAppSettings {
...
                     </FormRow>
                     <FormRow
                         i18nLabel = 'profileModal.startWithVideoMuted' >
                         <Switch
                             onValueChange = {
                                 this._onStartVideoMutedChange
                             }
-                            value = { this.state.startWithVideoMuted } />
+                            value = {
+                                this.props._profile.startWithVideoMuted
```
Comment (saghul) [inline react/features/app-settings/components/AppSettings.native.js]: general observation: I think it's worth it to deconstruct the props / profile to avoid long lines.

File: react/features/app-settings/components/BackButton.native.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    /**
+    * An external style object passed to the component.
+    */
+    style: Object
+};
+
+/**
+ * A component rendering a back button that looks native on both platforms.
+ */
+export default class BackButton extends Component<Props, *> {
```
Comment (saghul) [inline react/features/app-settings/components/BackButton.native.js]: ditto for the *

File: react/features/app-settings/components/BackButton.native.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+    /**
+     * Implements React's {@link Component#render()}, renders the button.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const cssPostfix = _.startCase(_.toLower(Platform.OS));
```
Comment (saghul) [inline react/features/app-settings/components/BackButton.native.js]: the OS is always given in lowercase

File: react/features/app-settings/components/BackButton.native.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+        return (
+            <TouchableOpacity
+                accessibilityLabel = { 'Back' }
+                onPress = { this.props.onPress }>
+                <Icon
+                    name = {
+                        Platform.OS === 'android'
```
Comment (saghul) [inline react/features/app-settings/components/BackButton.native.js]: make this a constant instead of computing it on every render

File: react/features/app-settings/components/FormRow.native.js [hunk 11]
```diff
@@ -39,7 +47,7 @@ type Props = {
  * Implements a React {@code Component} which renders a standardized row
  * on a form. The component should have exactly one child component.
  */
-class FormRow extends Component<Props> {
+class FormRow extends Component<Props, *> {
```
Comment (saghul) [inline react/features/app-settings/components/FormRow.native.js]: ditto

File: react/features/app-settings/components/styles.js [hunk 12]
```diff
@@ -4,32 +4,49 @@ import {
...
- * {@code WelcomePage} and {@code BlankPage}.
+ * The styles of the React {@code Components} of the feature
+ * {@code AppSettings}.
  */
 export default createStyleSheet({
 
+    /**
+    * The back button style on Android platform.
+    */
+    backIconAndroid: {
```
Comment (saghul) [inline react/features/app-settings/components/styles.js]: I think this could be simplified using `Platform.select`: https://facebook.github.io/react-native/docs/platform-specific-code.html

File: react/features/app-settings/functions.native.js [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+*/
+function _isIphoneX() {
+    const {
+        height: SCREEN_HEIGHT,
+        width: SCREEN_WIDTH
+    } = Dimensions.get('window');
+
+    return (
+        Platform.OS === 'ios'
+        && ((SCREEN_HEIGHT === IPHONEX_HEIGHT && SCREEN_WIDTH === IPHONEX_WIDTH)
```
Comment (saghul) [inline react/features/app-settings/functions.native.js]: style: these need to be indented one extra level

File: react/features/app-settings/functions.native.js [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+const IPHONEX_WIDTH = 375;
+
+/**
+* Determines if it's an iPhone X or not to handle notch on the settings
+* screen properly.
+*
+* FIXME: Settings screen uses a custom implementation of safe views
+* to avoid empty areas on the screen when there should be a header rendered.
+* In long term we need to use a generic solution to handle "safe view
+* areas" for various devices. It can be React 0.50's {@code SafeAreaView} or
```
Comment (saghul) [inline react/features/app-settings/functions.native.js]: I didn't know it already existed in RN! why are we not using it then? wouldn't it be much simpler than what we are doing here now?

**Comment by saghul** [comment]
@zbettenbuk the top padding seems incorrect on non-X iPhones in landscape mode:

img_0018

File: react/features/app-settings/components/AppSettings.native.js (hunk 4 above)
Comment (zbettenbuk) [reply react/features/app-settings/components/AppSettings.native.js]: It's needed for the underlying view to be re-layout'ed when rotated, so then this screen receives the aspect ration change. It doesn't work with "fullScreen"

File: react/features/app-settings/functions.native.js (hunk 14 above)
Comment (zbettenbuk) [reply react/features/app-settings/functions.native.js]: I posted a note on Stride about how much I don't like how it renders the screen (too much empty space). Let's talk about it later, when we implement something more generic

File: react/features/app-settings/components/AppSettings.native.js (hunk 4 above)
Comment (saghul) [reply react/features/app-settings/components/AppSettings.native.js]: I commented it out and it worked for me. Looks like `supportedOrientations` is the only required thing. Can you please double-check?

File: react/features/app-settings/functions.native.js (hunk 14 above)
Comment (saghul) [reply react/features/app-settings/functions.native.js]: ok

File: react/features/app-settings/components/AppSettings.native.js (hunk 4 above)
Comment (zbettenbuk) [reply react/features/app-settings/components/AppSettings.native.js]: Re-checked, the orientation does work, but the screen doesn't get re-layout'd and the paddings will be outdated. Please test it on an iPhone (the most visible it is on an iPhone X). Open the settings screen in portrait, then rotate the device. The header will still have the safety offset (where it shouldn't) and the side won't have it (while it should). Or is it only me?
image

Comment (saghul) [reply react/features/app-settings/components/AppSettings.native.js]: Isn't the aspectRatio a prop? If so, it would trigger a re-render.

Comment (zbettenbuk) [reply react/features/app-settings/components/AppSettings.native.js]: Yes, bit it doesn't change

Comment (saghul) [reply react/features/app-settings/components/AppSettings.native.js]: I tested this and (I guess because this is a Modal) our aspect ratio detector doesn't kick in, and thus there is no new props when we rotate. Sigh.

Comment (saghul) [reply react/features/app-settings/components/AppSettings.native.js]: I "solved" it as follows:

```
    componentWillMount() {
        this._orientationListener = DeviceEventEmitter.addListener(
            'namedOrientationDidChange', () => {
                console.log('ORIENTATION CHANGED');
                this.forceUpdate();
            },
        );
    }

    componentWillUnmount() {
        this._orientationListener.remove();
    }
```

I'm thinking we might want to use that for the aspect ratio detector...

File: react/features/app-settings/components/AppSettings.native.js (hunk 3 above)
Comment (zbettenbuk) [reply react/features/app-settings/components/AppSettings.native.js]: I insist 😆  I my world class definitions need an extra line after

File: react/features/app-settings/components/AppSettings.native.js (hunk 4 above)
Comment (zbettenbuk) [reply react/features/app-settings/components/AppSettings.native.js]: I leave it to your PR then to fix the detector as we agreed.

File: react/features/app-settings/components/AppSettings.native.js (hunk 3 above)
Comment (saghul) [reply react/features/app-settings/components/AppSettings.native.js]: Please look around and be consistent.

File: react/features/app-settings/components/AppSettings.native.js [hunk 15]
```diff
@@ -23,55 +28,76 @@ import { translate } from '../../base/i18n';
...
+
     /**
      * Implements React's {@link Component#render()}, renders the settings page.
      *
      * @inheritdoc
      * @returns {ReactElement}
      */
     render() {
-        const { t } = this.props;
+        const { _profile, t } = this.props;
```
Comment (saghul) [inline react/features/app-settings/components/AppSettings.native.js]: deconstructing _profile may also be worth it here

File: react/features/app-settings/functions.native.js [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+*/
+
+/**
+* Determines if the device is an iPad or not.
+*
+* @returns {boolean}
+*/
+export function isIPad() {
+    const {
+        height: SCREEN_HEIGHT,
```
Comment (saghul) [inline react/features/app-settings/functions.native.js]: why the capitalized values?

File: react/features/app-settings/functions.native.js [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+* areas" for various devices. It can be React 0.50's {@code SafeAreaView} or
+* a module like 'react-native-safe-area-view' or a custom implemented solution.
+*/
+
+/**
+* Determines if the device is an iPad or not.
+*
+* @returns {boolean}
+*/
+export function isIPad() {
```
Comment (saghul) [inline react/features/app-settings/functions.native.js]: this looks like something which belongs in base/react/platform

File: react/features/app-settings/functions.native.js [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+        / Math.min(SCREEN_HEIGHT, SCREEN_WIDTH)) < 1.6;
+}
+
+/**
+* Determines if it's an iPhone X or not to handle notch on the settings
+* screen properly.
+*
+* @returns {boolean}
+*/
+function _isIPhoneX() {
```
Comment (saghul) [inline react/features/app-settings/functions.native.js]: ditto

File: react/features/app-settings/functions.native.js [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+*/
+export function isIPad() {
+    const {
+        height: SCREEN_HEIGHT,
+        width: SCREEN_WIDTH
+    } = Dimensions.get('window');
+
+    return Platform.OS === 'ios' && (
+        Math.max(SCREEN_HEIGHT, SCREEN_WIDTH)
+        / Math.min(SCREEN_HEIGHT, SCREEN_WIDTH)) < 1.6;
```
Comment (saghul) [inline react/features/app-settings/functions.native.js]: style: one more indent level

File: react/features/app-settings/components/BackButton.native.js [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+    render() {
+        const isAndroid = Platform.OS === 'android';
+
+        return (
+            <TouchableOpacity
+                accessibilityLabel = { 'Back' }
+                onPress = { this.props.onPress }>
+                <Icon
+                    name = {
+                        isAndroid ? 'arrow_back' : 'navigate_before'
```
Comment (saghul) [inline react/features/app-settings/components/BackButton.native.js]: extract this computation to a constant at the top level, there is no reason to compute it every time

File: lang/main.json [hunk 21]
```diff
@@ -541,8 +541,8 @@
     "profileModal": {
         "displayName": "Display name",
         "email": "Email",
+        "defaultURL": "Server URL",
         "header": "Settings",
-        "serverURL": "Server URL",
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply lang/main.json]: Can't really see, what's the problem with it?

File: react/features/app-settings/components/AppSettings.native.js (hunk 15 above)
Comment (zbettenbuk) [reply react/features/app-settings/components/AppSettings.native.js]: I'm not sure it is. Reading the code, "profile.defaultURL" is something I understand immediately, while a variable simply named "defaultURL" may have multiple meaning.

File: react/features/app-settings/functions.native.js (hunk 17 above)
Comment (zbettenbuk) [reply react/features/app-settings/functions.native.js]: I think the comment applies this too: these pieces of logic (actually maybe the entire file) needs to be refactored when we have a generic SafeArea solution, and since no other feature uses this method yet, I'd leave it here now. Also, I'd still prefer a hardware-info based solution replacing these.

File: lang/main.json (hunk 21 above)
[the remark this answers is not in the document]
Comment (saghul) [reply lang/main.json]: all language files would need to be updated. let's leave this change out

[the remark this answers is not in the document]
Comment (paweldomas) [reply lang/main.json]: If we change the key then translation will be broken, because those translation will be for the old key. But it depends if this's been translated already. Given how fresh the profileModal is, I think we can change it.

File: react/features/app-settings/components/BackButton.native.js [hunk 22]
```diff
@@ -0,0 +1,54 @@
...
+
+import React, { Component } from 'react';
+import { TouchableOpacity } from 'react-native';
+
+import styles from './styles';
+
+import { Icon } from '../../base/font-icons';
+import { Platform } from '../../base/react';
+
+const ICON = Platform.OS === 'android' ? 'arrow_back' : 'navigate_before';
```
Comment (saghul) [inline react/features/app-settings/components/BackButton.native.js]: please add a comment

File: react/features/app-settings/components/AppSettings.native.js [hunk 23]
```diff
@@ -30,48 +45,58 @@ class AppSettings extends AbstractAppSettings {
...
+                    'portrait'
+                ] }
                 visible = { this.props._visible }>
-                <View style = { styles.headerContainer } >
+                <View
+                    style = { [
+                        styles.headerContainer,
+                        this._getSafetyPadding()
+                    ] } >
+                    <BackButton
```
Comment (paweldomas) [inline react/features/app-settings/components/AppSettings.native.js]: is the Android hardware back button handled by the profile settings screen ?

Comment (zbettenbuk) [reply react/features/app-settings/components/AppSettings.native.js]: Yes

File: react/features/app-settings/components/BackButton.native.js (hunk 22 above)
Comment (saghul) [reply react/features/app-settings/components/BackButton.native.js]: also a more descriptive name such as BACK_ICON_NAME sounds more appropriate IMHO

File: react/features/app-settings/functions.native.js (hunk 17 above)
Comment (saghul) [reply react/features/app-settings/functions.native.js]: There are other reasons why we might want to test for a specific device. This is related to the platform itself, so it certainly belongs there. If we end up removing it great, if not, it will already be in the right place. Since this code is being added now, let's add it in the right place.

File: react/features/app-settings/functions.native.js [hunk 24]
```diff
@@ -0,0 +1,73 @@
...
+
+/**
+* Determines if it's an iPhone X or not to handle notch on the settings
+* screen properly.
+*
+* @returns {boolean}
+*/
+function _isIPhoneX() {
+    const {
+        height: screenHeight,
```
Comment (saghul) [inline react/features/app-settings/functions.native.js]: I don't think the rename adds anything here, it's used right below and the context is clear enough.

File: lang/main.json (hunk 21 above)
[the remark this answers is not in the document]
Comment (paweldomas) [reply lang/main.json]: @saghul currently it's only in main.json, but I don't know which one is better so I don't mind if it stays

[the remark this answers is not in the document]
Comment (saghul) [reply lang/main.json]: Yeah, see my other comment though.

[the remark this answers is not in the document]
Comment (paweldomas) [reply lang/main.json]: Yes, it's the server URL. "default URL" feels more like a whole thing including room name or whatever.

File: react/features/app-settings/functions.native.js (hunk 17 above)
Comment (paweldomas) [reply react/features/app-settings/functions.native.js]: I would also vote for moving outside. Nobody will know that this thing exists here and will duplicate it eventually.

File: react/features/base/react/functions.native.js [hunk 25]
```diff
@@ -0,0 +1,37 @@
...
+
+const IPHONEX_HEIGHT = 812;
+const IPHONEX_WIDTH = 375;
+
+/**
+* Determines if the device is an iPad or not.
+*
+* @returns {boolean}
+*/
+export function isIPad() {
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/base/react/functions.native.js]: Based on this page, yes: https://www.lifewire.com/list-of-ipad-models-and-generations-1994232

Btw there is a threshold, calculating that, as all iPad models (so far) has a ratio 1:1.33, so 1.6 seems to be a safe limit (with status bar and everything included).

**Comment by zbettenbuk** [comment]
@saghul thanks, fixed all, left comment on one.

About the section headers: yes, Lyubo suggested to have something, but the styling is mine, so that may need additional review from @yanas

File: react/features/base/react/functions.native.js (hunk 25 above)
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/base/react/functions.native.js]: Ack

**Comment by saghul** [comment]
> About the section headers: yes, Lyubo suggested to have something, but the styling is mine, so that may need additional review from @yanas

No problem, we can improve that a bit later, I just wanted to know where it came from 👍

**Comment by zbettenbuk** [comment]
@saghul I'm checking the tests, but weird, they are. The same test fails for Hristo's other PR.

**Comment by saghul** [comment]
@damencho ping, can you help :-)

**Comment by damencho** [comment]
Yep, just disabled the test and will look at it later, apparently, there is some problem between it and FF.
Also edited the template index file to have the correct link to chrome-2-chrome tests and to firefox-2-chrome tests, will take effect on new ran PR tests.

File: react/features/base/profile/middleware.js [hunk 26]
```diff
@@ -2,7 +2,10 @@
 import { PROFILE_UPDATED } from './actionTypes';
 import MiddlewareRegistry from '../redux/MiddlewareRegistry';
```
Comment (lyubomir) [inline react/features/base/profile/middleware.js]: Please notice that we're importing from the redux feature in multiple lines!

````javascript
import MiddlewareRegistry from '../redux/MiddlewareRegistry';
import { toState } from '../redux';
````

That's why we group imports in the order third-party, other/outside feature, this/inside feature and separate the groups with empty lines. Let's please follow the coding style!

I'm taking care of this one.

File: react/features/base/profile/middleware.js [hunk 27]
```diff
@@ -31,13 +34,11 @@ MiddlewareRegistry.register(store => next => action => {
  * @returns {void}
  */
 function _updateLocalParticipant(store) {
+    const localParticipant = getLocalParticipant(toState(store));
     const profile = getProfile(toState(store));
```
Comment (lyubomir) [inline react/features/base/profile/middleware.js]: Please note that `toState(store)` is executed twice in the two lines above! That's very likely the result of copy-pasting that that's why copy-pasting is discouraged by our coding style.

I'm taking care of this one.

File: react/features/base/profile/middleware.js [hunk 28]
```diff
@@ -31,13 +34,11 @@ MiddlewareRegistry.register(store => next => action => {
...
 
-    const newLocalParticipant = {
-        email: profile.email,
-        local: true,
-        name: profile.displayName
-    };
+    localParticipant.email = profile.email;
+    localParticipant.name = profile.displayName;
```
Comment (lyubomir) [inline react/features/base/profile/middleware.js]: Ouch! Did we just modify the redux state here?

File: react/features/base/react/functions.native.js [hunk 29]
```diff
@@ -0,0 +1,37 @@
...
+            / Math.min(height, width)) < 1.6;
+}
+
+/**
+* Determines if it's an iPhone X or not.
+*
+* @returns {boolean}
+*/
+export function isIPhoneX() {
+    const { height, width } = Dimensions.get('window');
```
Comment (lyubomir) [inline react/features/base/react/functions.native.js]: I don't like `isIPad` and `isIPhoneX`. We're in freaking react-native which allows us to write native source code to query iOS what the device is and we're writing some obscure source code involving the app's window dimensions and aspect rations?! Why are we OK with it, @saghul?

Comment (zbettenbuk) [reply react/features/base/react/functions.native.js]: Ok, so two things to explain here:

- The point of using an implementation like this is to minimise the blank area around screens while properly handle the iPhoneX top notch and iPhone's status bar (that behaves differently on iPad). I could have used React's new SafeArea component, but with screens that have a header (or similar, native looking screens) it renders a bad looking empty area around, that this implementation avoids.
- In another discussion with Saul we talked about introducing a new native module that would do similar things by getting these info from the device but it turned out that it doesn't solve all our problems so we decided not to introduce a new native module if it only gives us half of the solution.

File: react/features/base/profile/middleware.js (hunk 28 above)
Comment (zbettenbuk) [reply react/features/base/profile/middleware.js]: Got the feedback to do it like this - assumed that I was asked to do this since this is only a copy of the store (or was it your point back then @saghul ? not entirely sure because we didn't discuss it). Originally the object was deconstructed and a new instance created (see above).

File: react/features/app-settings/components/AbstractAppSettings.js [hunk 30]
```diff
@@ -10,6 +10,16 @@ import { getProfile, updateProfile } from '../../base/profile';
 */
 type Props = {
 
+    /**
+    * The current aspect ratio of the screen.
+    */
```
Comment (lyubomir) [inline react/features/app-settings/components/AbstractAppSettings.js]: I'd like to please attract your attention to the formatting of the multiline comments. I presume it's an editor setting so could you please make sure you comply with our coding style?

We don't do:
````javascript
/**
*
*/
````

We do:
````javascript
/**
 *
 */
````

File: react/features/base/profile/middleware.js (hunk 28 above)
Comment (saghul) [reply react/features/base/profile/middleware.js]: Probably misread, my bad! Thanks for catching that @lyubomir !

File: react/features/base/react/functions.native.js (hunk 29 above)
Comment (saghul) [reply react/features/base/react/functions.native.js]: We made an attempt to use SafeArea as is, but IMHO it needs to be a larger change which other components such as WelcomePage and Conference also use, so I conceded going this route since the broken code was already in.
